test/martha/martha_v3.test.js

```diff
@@ -424,412 +382,332 @@ test.serial('martha_v3 returns an error when an invalid field is requested', asy
-    const saKeyUrlMatches = requestedBondSAKeyUrl.match(bondSAKeyUrlRegEx);
-    t.truthy(saKeyUrlMatches, 'Bond SA key URL called does not match Bond SA key URL regular expression');
-    const expectedSAKeyProvider = 'dcf-fence';
-    const actualSAKeyProvider = saKeyUrlMatches[2];
-    t.is(actualSAKeyProvider, expectedSAKeyProvider);
+    sinon.assert.callCount(getJsonFromApiStub, 2);
 });
 
 test.serial('martha_v3 calls bond Bond with the "fence" provider when the Data Object URL host is "dg.4503"', async (t) => {
-    getJsonFromApiStub.onCall(0).resolves(sampleDosResponse);
+    const bond = bondUrls('fence');
+    const drs = drsUrls(config.HOST_BIODATA_CATALYST_STAGING);
+    getJsonFromApiStub.withArgs(bond.serviceAccountKeyUrl, terraAuth).resolves(googleSAKeyObject);
+    getJsonFromApiStub.withArgs(drs.objectsUrl, null).resolves(bdcDrsResponse);
     const response = mockResponse();
+
     await marthaV3(mockRequest({ body: { 'url': 'drs://dg.4503/this_part_can_be_anything' } }), response);
-    const requestedBondSAKeyUrl = getJsonFromApiStub.getCall(1).args[0];
-    const saKeyUrlMatches = requestedBondSAKeyUrl.match(bondSAKeyUrlRegEx);
-    t.truthy(saKeyUrlMatches, 'Bond SA key URL called does not match Bond SA key URL regular expression');
-    const expectedSAKeyProvider = 'fence';
-    const actualSAKeyProvider = saKeyUrlMatches[2];
-    t.is(actualSAKeyProvider, expectedSAKeyProvider);
-});
-
-test.serial(
-    'martha_v3 does call Bond and return SA key when the host url is for dataguids.org',
-    async (t) => {
-        getJsonFromApiStub.onCall(0).resolves(dataGuidsOrgResponse);
-        getJsonFromApiStub.onCall(1).resolves(googleSAKeyObject);
-        const response = mockResponse();
-        await marthaV3(
-            mockRequest({ body: { 'url': 'dos://dataguids.org/a41b0c4f-ebfb-4277-a941-507340dea85d' } }),
-            response
-        );
-        t.is(response.statusCode, 200);
-        const result = response.send.lastCall.args[0];
-        sinon.assert.callCount(getJsonFromApiStub, 2); // Bond was called to get SA key
-        t.deepEqual({ ...result }, dataGuidsOrgMarthaResult(googleSAKeyObject));
-        t.is(
-            getJsonFromApiStub.getCall(0).args[0],
-            'https://dataguids.org/ga4gh/dos/v1/dataobjects/a41b0c4f-ebfb-4277-a941-507340dea85d',
-        );
-        t.falsy(getJsonFromApiStub.getCall(0).args[1]); // no auth passed
-        const requestedBondSAKeyUrl = getJsonFromApiStub.getCall(1).args[0];
-        const saKeyUrlMatches = requestedBondSAKeyUrl.match(bondSAKeyUrlRegEx);
-        t.truthy(saKeyUrlMatches, 'Bond SA key URL called does not match Bond SA key URL regular expression');
-        const expectedSAKeyProvider = 'dcf-fence';
-        const actualSAKeyProvider = saKeyUrlMatches[2];
-        t.is(actualSAKeyProvider, expectedSAKeyProvider);
-    }
-);
+
+    t.is(response.statusCode, 200); // Not strictly necessary here, but the test fails if we don't assert something
```

Comment:
This also makes me feel like this test overlaps with other tests. I'm leaving it here because I'm envisioning a future where this case could be checked in a unit test suite (no mocks/stubs) and its presence will remind us that this needs to be covered in that test suite.